tests/test_caley_transform.py

```diff
@@ -0,0 +1,91 @@
+import torch
+import random
+import numpy as np
+import pytest
+import geoopt.manifolds.siegel.csym_math as sm
+from geoopt.linalg import sym
+from geoopt import UpperHalf, BoundedDomain
+
+
+@pytest.fixture(scope="module", autouse=True, params=range(30, 40))
+def seed(request):
+    seed = request.param
+    torch.manual_seed(seed)
+    random.seed(seed)
+    np.random.seed(seed)
+    return seed
+
+
+@pytest.fixture(scope="module", params=[torch.complex128, torch.complex64])
```

Comment:
maybe add float dtype to the test suit?

Reply:
We use the Cayley Transform to move from one model (Upper half or Bounded domain) to the other and vice versa, thus we only define it for complex matrices.